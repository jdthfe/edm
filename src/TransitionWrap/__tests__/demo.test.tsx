import webDemoTest from '@tests/shared/demoTest';
webDemoTest('TransitionWrap');

import React from 'react';
import Demo from '../demo';
// import { transitionFade } from '../../_util/variable';
import { render, fireEvent, waitForElement } from 'react-testing-library';

describe('TransitionWrapDemo', () => {
    it('animation', async () => {
        const { baseElement, getAllByRole } = render(<Demo />);

        // https://testing-library.com/docs/dom-testing-library/cheatsheet#queries
        fireEvent.click(getAllByRole('button')[0]);
        // fireEvent.click(getAllByRole('button')[1]);
        // fireEvent.click(getAllByRole('button')[2]);

        // https://testing-library.com/docs/dom-testing-library/api-async
        // await act(async () => {
        await waitForElement(() => baseElement.querySelector(`.hello-entry`));
        await waitForElement(() => baseElement.querySelector(`.hello-entry-active`));
        await waitForElement(() => baseElement.querySelector(`.hello-entry-done`));

        expect(baseElement).toMatchSnapshot('visible=true');
        fireEvent.click(getAllByRole('button')[0]);

        await waitForElement(() => baseElement.querySelector(`.hello-exit`));
        await waitForElement(() => baseElement.querySelector(`.hello-exit-active`));
        // await waitForElement(() => baseElement.querySelector(`.hello-exit-done`));

        expect(baseElement).toMatchSnapshot('visible=false');
    });
});
