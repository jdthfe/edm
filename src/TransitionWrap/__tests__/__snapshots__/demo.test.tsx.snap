// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransitionWrapDemo animation: visible=false 1`] = `
<body>
  <div>
    <div
      class="edm-wingblank TransitionWrap edm-wingblank-m"
    >
      <button
        class="edm-button edm-button-l edm-button-customize edm-button-ghost"
        role="button"
      >
        click to 
        show
         text
      </button>
      <h1
        class="hello hello-exit-active"
      >
        hello
      </h1>
      <h1
        class="world world-exit-active"
      >
        world
      </h1>
    </div>
  </div>
</body>
`;

exports[`TransitionWrapDemo animation: visible=true 1`] = `
<body>
  <div>
    <div
      class="edm-wingblank TransitionWrap edm-wingblank-m"
    >
      <button
        class="edm-button edm-button-l edm-button-customize edm-button-ghost"
        role="button"
      >
        click to 
        hide
         text
      </button>
      <h1
        class="hello hello-entry-done"
      >
        hello
      </h1>
      <h1
        class="world world-entry-done"
      >
        world
      </h1>
    </div>
  </div>
</body>
`;

exports[`renders TransitionWrap correctly 1`] = `
<div>
  <div
    class="edm-wingblank TransitionWrap edm-wingblank-m"
  >
    <button
      class="edm-button edm-button-l edm-button-customize edm-button-ghost"
      role="button"
    >
      click to 
      show
       text
    </button>
    <h1
      class="world world-exit-done"
    >
      world
    </h1>
  </div>
</div>
`;
